lib/charms/mongodb/v0/set_status.py

```diff
@@ -105,20 +105,17 @@ def share_status_to_config_server(self):
 
     def is_unit_status_ready_for_upgrade(self) -> bool:
         """Returns True if the status of the current unit reflects that it is ready for upgrade."""
-        current_status = type(self.charm.unit.status)
-        status_message = self.charm.unit.status.message
+        current_status = self.charm.unit.status
+        status_message = current_status.message
         if isinstance(current_status, ActiveStatus):
             return True
 
         if not isinstance(current_status, WaitingStatus):
             return False
 
-        if (
-            status_message
-            and status_message != Config.Status.CONFIG_SERVER_WAITING_FOR_REFRESH.message
-        ):
-            return False
+        if status_message and "is not up-to date with config-server" in status_message:
```

Comment:
looking for `CONFIG_SERVER_WAITING_FOR_REFRESH ` was incorrect

Reply:
@MiaAltieri can we have it as a constant instead of just a string? Also update the place where the charm message is set to use this constant.

Reply:
We could but the actual message is much longer - it includes the revision numbers of both the config-server and the shard, so doing
>  update the place where the charm message is set to use this constant.

would actually be more complicated than just that constant